### PR TITLE
[addons] fix: add missing nullptr check in installer

### PR DESCRIPTION
**Pull request by emveepee**
## Description

PR 18439 https://github.com/xbmc/xbmc/pull/18439 introduced a crash on install when and addon is being installed from a zip file with dependencies that are found on a repo that also need to be installed.

This change adds a test to only use a repo value when the source is a repo

## Motivation and Context

Kodi should not crash installing addons from zip files

This was discussed with @howie-f on the Kodi Forum https://forum.kodi.tv/showthread.php?tid=357269&action=lastpost

## How Has This Been Tested?

Tested using a zipped addon which worked prior to PR 18349 and doesn't crash with the fix

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/AddonInstaller.cpp [hunk 1]
```diff
@@ -818,10 +818,11 @@ bool CAddonInstallJob::Install(const std::string &installFrom, const RepositoryP
           AddonPtr dependencyToInstall;
 
           // origin of m_addon is empty at least if an addon is installed for the first time
-          // we need to override "parentRepoId" in this case
+          // we need to override "parentRepoId" in this case when the addon is installed
+          // from a repo
```
Comment (howie-f) [inline xbmc/addons/AddonInstaller.cpp:822]: maybe `we need to override "parentRepoId" if the passed in repo is valid`. this error didn't happen when installed from a repo i think

File: xbmc/addons/AddonInstaller.cpp [hunk 2]
```diff
@@ -818,10 +818,11 @@ bool CAddonInstallJob::Install(const std::string &installFrom, const RepositoryP
...
 
           const std::string& parentRepoId =
-              m_addon->Origin().empty() ? repo->ID() : m_addon->Origin();
+              m_addon->Origin().empty() && repo? repo->ID() : m_addon->Origin();
```
Comment (howie-f) [inline xbmc/addons/AddonInstaller.cpp:825]: please add a space between `repo` and `?` to satisfy clang-format

**Review by howie-f — COMMENTED** [review]
as suggestion i would change pr title and commit message to something like
`[addons] fix: add missing nullptr check in installer`

File: xbmc/addons/AddonInstaller.cpp (hunk 1 above)
Comment (emveepee) [reply xbmc/addons/AddonInstaller.cpp]: Done

File: xbmc/addons/AddonInstaller.cpp (hunk 2 above)
Comment (emveepee) [reply xbmc/addons/AddonInstaller.cpp]: Done

**Comment by howie-f** [comment]
jenkins build this please

**Review by howie-f — APPROVED** [review]
really good catch. looks fine to me :+1:

**Review by phunkyfish — APPROVED** [review]

**Comment by howie-f** [comment]
Thanks for your contribution @emveepee !

**Pull request merged**